qualtran/bloqs/arithmetic/multiplication.py

```diff
@@ -12,16 +12,78 @@
 #  See the License for the specific language governing permissions and
 #  limitations under the License.
 
-from typing import Set, TYPE_CHECKING
+from typing import Any, Dict, Iterable, Sequence, Set, TYPE_CHECKING, Union
 
+import cirq
 from attrs import frozen
 
-from qualtran import Bloq, Register, Side, Signature
+from qualtran import Bloq, GateWithRegisters, Register, Side, Signature
 from qualtran.bloqs.basic_gates import Toffoli
 from qualtran.cirq_interop.t_complexity_protocol import TComplexity
 
 if TYPE_CHECKING:
     from qualtran.resource_counting import BloqCountT, SympySymbolAllocator
+    from qualtran.simulation.classical_sim import ClassicalValT
+
+
+@frozen
+class PlusEqualProduct(GateWithRegisters, cirq.ArithmeticGate):
```

Comment:
what does `cirq.ArithmeticGate` buy us here?

Reply:
Ability to do (a) easily define a cirq-style unitary / simulations behavior and b) perform cirq-style simulations using a faster algorithm implemented in the `_apply_unitary_` method on `ArithmeticGate`

Reply:
Let's bias towards the classical simulation support in qualtran rather than cirq